pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy

```diff
@@ -194,13 +196,13 @@ public class Converter {
     }
 
     /**
-     * Converts the raw script from a {@link CpsScript} into {@link ModelASTPipelineDef}
+     * Converts the raw script from a {@link WorkflowRun} into {@link ModelASTPipelineDef}
      *
-     * @param cpsScript The {@link CpsScript} to pull from.
+     * @param run The {@link WorkflowRun} to pull from.
      * @return A parsed and validated {@link ModelASTPipelineDef}
      */
-    public static ModelASTPipelineDef parseFromCpsScript(CpsScript cpsScript) {
-        CpsFlowExecution execution = (CpsFlowExecution) FieldUtils.readField(cpsScript, "execution", true)
+    public static ModelASTPipelineDef parseFromWorkflowRun(WorkflowRun run) {
```

Comment:
If anyone is seriously treating this as an API, they deserve to have their plugin broken.

Really I would like to introduce a system like OSGi has (NetBeans modules, Java 9 modules, …) whereby we can just declare _all_ types in the plugin to be off limits unless and until explicitly exported to the world.